codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -183,28 +203,32 @@ public void incrementWindowSize(ChannelHandlerContext ctx, Http2Stream stream, i
             // Update the stream window and write any pending frames for the stream.
             AbstractState state = state(stream);
             state.incrementStreamWindow(delta);
-            state.writeBytes(state.writableWindow());
+            if (isWritable()) {
+                state.writeBytes(state.writableWindow());
```

Comment:
Currently this PR does not change the write order (which is not done in priority order).  Depending on what we decide for https://github.com/netty/netty/pull/3864#issuecomment-110151715 the granularity of checking can be increased, but write order will likely have to change.